src/components/grid/grid.ts

```diff
@@ -656,6 +676,105 @@ export class Grid extends Component<Props, SpreadsheetChildEnv> {
     this.menuState.isOpen = false;
     this.focus();
   }
+
+  private processHeaderGroupingKey(direction: Direction) {
+    if (this.env.model.getters.getSelectedZones().length !== 1) {
+      return;
+    }
+
+    const selectingRows = this.env.model.getters.getActiveRows().size > 0;
+    const selectingCols = this.env.model.getters.getActiveCols().size > 0;
+
+    if (selectingCols && selectingRows) {
+      this.processHeaderGroupingEventOnWholeSheet(direction);
+    } else if (selectingCols) {
+      this.processHeaderGroupingEventOnHeaders(direction, "COL");
+    } else if (selectingRows) {
+      this.processHeaderGroupingEventOnHeaders(direction, "ROW");
+    } else {
+      this.processHeaderGroupingEventOnGrid(direction);
+    }
+  }
+
+  private processHeaderGroupingEventOnHeaders(direction: Direction, dimension: Dimension) {
+    const sheetId = this.env.model.getters.getActiveSheetId();
+
+    const zone = this.env.model.getters.getSelectedZone();
+    const start = dimension === "COL" ? zone.left : zone.top;
+    const end = dimension === "COL" ? zone.right : zone.bottom;
+
+    switch (direction) {
+      case "right":
+        this.env.model.dispatch("GROUP_HEADERS", { sheetId, dimension: dimension, start, end });
+        break;
+      case "left":
+        this.env.model.dispatch("UNGROUP_HEADERS", { sheetId, dimension: dimension, start, end });
+        break;
+      case "down":
+        this.env.model.dispatch("EXPAND_HEADER_GROUPS_IN_ZONE", { sheetId, dimension, zone });
+        break;
+      case "up":
+        this.env.model.dispatch("COLLAPSE_HEADER_GROUPS_IN_ZONE", { sheetId, dimension, zone });
+        break;
+    }
+  }
+
+  private processHeaderGroupingEventOnWholeSheet(direction: Direction) {
+    const sheetId = this.env.model.getters.getActiveSheetId();
+    if (direction === "up") {
+      this.env.model.dispatch("COLLAPSE_ALL_HEADER_GROUPS", { sheetId, dimension: "ROW" });
+      this.env.model.dispatch("COLLAPSE_ALL_HEADER_GROUPS", { sheetId, dimension: "COL" });
+    } else if (direction === "down") {
+      this.env.model.dispatch("EXPAND_ALL_HEADER_GROUPS", { sheetId, dimension: "ROW" });
```

Comment:
naming detail: maybe name those commands `FOLD_*` and `UNFOLD_*` to be consistent with the `isFolded` property used a bit everywhere?